src/NetMQ/NetMQPoller.cs

```diff
@@ -120,25 +120,63 @@ protected override void QueueTask(Task task)
             m_tasksQueue.Enqueue(task);
         }
 
+        internal Task RunAsync([NotNull] Action action)
+        {
+            Task t;
+
+            if (!IsRunning || CanExecuteTaskInline)
+            {
+                action();
+
+                t = FromResult<object>(null);
+            }
+            else
+            {
+                t = Task.Factory.StartNew(action);
+            }
+
+            return t;
+        }
+
         /// <summary>
         /// Run an action on the Poller thread
         /// </summary>
         /// <param name="action">The action to run</param>
+        [Obsolete("Queues the action on the poller's thread, but provides no sync mechanism.  Please use RemoveAsync() to remove sockets or timers")]
         public void Run([NotNull] Action action)
         {
             if (!IsRunning || CanExecuteTaskInline)
                 action();
             else
                 new Task(action).Start(this);
```

Comment:
I think this should use `Task.Factory.StartNew`.

Reply:
This is identical to the code in the original `Run(Action)` method.  See [here](https://github.com/zeromq/netmq/blob/455f6b066c6a775f91e51e4a2c88879bbe2b86c5/src/NetMQ/NetMQPoller.cs#L146).

Using `Task.Factory.StartNew` requires several additional parameters in addition to specifying the scheduler: 

[`Task<TResult> StartNew<TResult>(Func<TResult> function, CancellationToken cancellationToken, TaskCreationOptions creationOptions, TaskScheduler scheduler)`](https://docs.microsoft.com/en-us/dotnet/api/system.threading.tasks.taskfactory.startnew?view=netframework-4.8#System_Threading_Tasks_TaskFactory_StartNew_System_Action_System_Threading_CancellationToken_System_Threading_Tasks_TaskCreationOptions_System_Threading_Tasks_TaskScheduler_)

Certainly possible, but I'm uncertain what the `TaskCreationOptions` should be here, nor do we supply a `CancellationToken`.

Adding a `cancellationToken` parameter would break the existing API, and defining one requires creating it even though  it won't be used to stop the task, unless an arbitrary timeout is assigned.

Reply:
@drewnoakes I need further clarification here.   This request seems to require creating an API change?  The current `Task(action).Start(this)` ensures that the task is run on the poller thread.

Is there a compatibility issue with the current impl?

Reply:
Ahh, I see.  There is possibly some confusion in using `Task(action).Start(this)` vs. using the locally defined `Run(action)` method [here](https://github.com/zeromq/netmq/blob/e86706ffb9138a25b3bc0e10484efe8b89a64819/src/NetMQ/NetMQPoller.cs#L482).

These are functionally equivalent, but I can update this code to use the local `Run(action)` method for consistency?